### PR TITLE
Add Auto-Toggle and Send Scoreboard/Tab Menu Data

### PR DESCRIPTION
**Pull request by HackedOS**

### CODE REVIEW

**Review by Ekwav — APPROVED** [review]

**Comment by HackedOS** [comment]
Closes https://github.com/Coflnet/SkyblockMod/issues/39

**Comment by HackedOS** [comment]
Closes https://github.com/Coflnet/SkyblockMod/issues/43

File: src/main/java/de/torui/coflsky/EventRegistry.java [hunk 1]
```diff
@@ -207,4 +207,15 @@ public void OnGuiClick(GuiScreenEvent.MouseInputEvent mie) {
...
 		de.torui.coflsky.CountdownTimer.onRenderTick(event);
 	}
+
+	int UpdateThisTick = 0;
+	@SubscribeEvent(priority = EventPriority.LOWEST)
+	public void onTick(TickEvent.ClientTickEvent event) {
+		UpdateThisTick++;
+		if (UpdateThisTick >= 200) UpdateThisTick = 0;
+		if (UpdateThisTick == 0) {
+			ScoreboardData();
```
Comment (Ekwav) [inline src/main/java/de/torui/coflsky/EventRegistry.java:217]: Run this on the thread pool as well

File: src/main/java/de/torui/coflsky/EventRegistry.java [hunk 2]
```diff
@@ -123,26 +132,17 @@ public static String ExtractUuidFromInventory(IInventory inventory) {
...
-				ad.setItemId(last.second().first());
-				
-				if((LastViewAuctionInvocation+60*1000) >=  System.currentTimeMillis()) {
-					ad.setAuctionId(LastViewAuctionUUID);
-				} else {
-					ad.setAuctionId("");
+		if(CoflSky.Wrapper.isRunning) {
+			chatThreadPool.submit(() -> {
+				String msg = sce.message.getUnformattedText();
+				Pattern pattern = Pattern.compile(Configuration.getInstance().chatRegex, Pattern.CASE_INSENSITIVE);
```
Comment (Ekwav) [inline src/main/java/de/torui/coflsky/EventRegistry.java:138]: Please avoid recompilation by reusing the pattern

**Review by Ekwav — CHANGES_REQUESTED** [review]

File: src/main/java/de/torui/coflsky/EventRegistry.java [hunk 3]
```diff
@@ -39,7 +39,9 @@
 import static de.torui.coflsky.EventHandler.TabMenuData;
 
 public class EventRegistry {
+	Pattern chatpattern = Pattern.compile(Configuration.getInstance().chatRegex, Pattern.CASE_INSENSITIVE);
```
Comment (Ekwav) [inline src/main/java/de/torui/coflsky/EventRegistry.java:42]: problem is that this doesn't refresh now anymore.  
It should be compiled when loaded and when `privacySettings` is called from the server

**Review by Ekwav — APPROVED** [review]

**Pull request merged**